opentelemetry-api/src/opentelemetry/util/types.py

```diff
@@ -15,5 +15,7 @@
 
 import typing
 
-AttributeValue = typing.Union[str, bool, float]
+AttributeValue = typing.Union[
```

Comment:
This change belongs in a separate PR.

Reply:
I thought this would belong here since the goal of this PR is to validate all AttributeValues are a valid type. Part of implementing correct validation is making sure all types documented in the OT spec are considered valid. So this change seems relevant to this PR IMO. 

Why do you think this change belongs in a separate PR?

Reply:
Logically, these are two independent things:

1. Correcting the definition of AttributeValue to match the spec
2. Adding runtime-validation to ensure that runtime values used as attributes have the correct type.

Although of course, changing 1 does imply adapting 2. I gues it might be fine to leave the changes in one PR, since the change for 1 is really just a single line, so it may add to much overhead. But then please adapt the PR title to state that you are changing the definition of AttributeValue and not just validating existing types.

Reply:
I agree with @Oberon00 here - I think it's simpler and cleaner to have both items as separated items.

Reply:
I split out the change into #368